api/client/build.go

```diff
@@ -223,19 +234,19 @@ func (cli *DockerCli) CmdBuild(args ...string) error {
 		Remove:         *rm,
 		ForceRemove:    *forceRm,
 		PullParent:     *pull,
-		Isolation:      *isolation,
 		CPUSetCPUs:     *flCPUSetCpus,
 		CPUSetMems:     *flCPUSetMems,
 		CPUShares:      *flCPUShares,
 		CPUQuota:       *flCPUQuota,
 		CPUPeriod:      *flCPUPeriod,
 		CgroupParent:   *flCgroupParent,
-		ShmSize:        *flShmSize,
 		Dockerfile:     relDockerfile,
+		ShmSize:        shmSize,
 		Ulimits:        flUlimits.GetList(),
-		BuildArgs:      flBuildArg.GetAll(),
 		AuthConfigs:    cli.configFile.AuthConfigs,
 	}
+	options.IsolationLevel = container.IsolationLevel(*isolation)
```

Comment:
I don't think this warrants embedding. Also, setting the parameter in the struct literal is orthogonal to the embedding question.

I am not sure I understand why renaming a field would make it easier to have it embedded :S